evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -676,6 +692,39 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> CheatcodeStackExecutor<'a, 'b, B, P>
                 };
                 self.state_mut().expected_emits.push(expected_emit);
             }
+            HEVMCalls::MockCall(inner) => {
+                let calldata = inner.1.to_vec();
+                if calldata.len() < 4 {
+                    return evm_error("Calldata for mocked call must be at least 4 bytes")
+                }
+
+                let selector: [u8; 4] =
+                    calldata[0..=3].try_into().expect("calldata should always be at least 4 bytes");
+                let mocked_call = if calldata.len() == 4 {
+                    MockedCall::Selector(inner.2.to_vec())
+                } else {
+                    MockedCall::SelectorAndData(calldata, inner.2.to_vec())
+                };
+                self.state_mut().mocked_calls.insert((inner.0, selector), mocked_call);
+            }
+            HEVMCalls::ClearMockedCalls(_) => {
+                self.state_mut().mocked_calls = Default::default();
+            }
+            HEVMCalls::ExpectCall(inner) => {
+                let calldata = inner.1.to_vec();
+                if calldata.len() < 4 {
+                    return evm_error("Calldata for mocked call must be at least 4 bytes")
```

Comment:
I don't think this is correct. Expected calls should support fallback methods as well. In general, i would advise against usage of the concept of selectors at this level of the evm as we want to be agnostic as possible to the language people are writing in, and 4 byte selectors is entirely a solidity construct

This comment applies to most of this implementation

Reply:
makes sense

Reply:
hmmm the point of the short version is to support someone not having to know the input data, correct? i would look at just slicing off any extra data on the match. See my comments that on other parts of code that im going to write